src/analysis/retail/priest/shadow/modules/tier/Tier31ShadowPriest4Set.tsx

```diff
@@ -0,0 +1,191 @@
+import SPELLS from 'common/SPELLS';
+import TALENTS from 'common/TALENTS/priest';
+import { TIERS } from 'game/TIERS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, {
+  ApplyBuffEvent,
+  ApplyBuffStackEvent,
+  RefreshBuffEvent,
+  DamageEvent,
+  RemoveBuffEvent,
+} from 'parser/core/Events';
+import Abilities from 'parser/core/modules/Abilities';
+import EventHistory from 'parser/shared/modules/EventHistory';
+import SpellUsable from 'parser/shared/modules/SpellUsable';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import { calculateEffectiveDamage } from 'parser/core/EventCalculateLib';
+import ItemDamageDone from 'parser/ui/ItemDamageDone';
+
+class Tier31ShadowPriest4Set extends Analyzer {
+  static dependencies = {
+    abilities: Abilities,
+    eventHistory: EventHistory,
+    spellUsable: SpellUsable,
+  };
+  protected abilities!: Abilities;
+  protected eventHistory!: EventHistory;
+  protected spellUsable!: SpellUsable;
+  has4Piece: boolean = true;
+
+  stacks = 0; //current stacks of buff
+  recentStacks = 0; // stacks used to calculate damage, since stacks are sometimes removed before damage event
+  totalStacks = 0; //total Possible Stacks
+
+  damageSC = 0; //damage to Shadow Crash
+  stacksSC = 0; //Stacks spent on Shadow Crash
+
+  castSWP = false; //this tells if a cast of shadow word Pain has happened
+  stacksSWP = 0; //Stacks spent on Shadow Word Pain
+  damageSWP = 0; //damage to Shadow Word Pain
+
+  castSC = false; //this tells if a cast of Shadow Crash has happened
+  recentSCTimestamp = 0; //This is the time of the first instance of shadowcrash damage to a target from one cast.
+  bonusSWP = 2.25; // multiplier per stack for Shadow Word Pain
+  bonusSC = 0.5; // multiplier per stack for Shadow Crash
+
+  recentApplied = false; //the order of the events requires this for edge cases.
+  timestampOfLast = 0; //Used to calculate unused buffs
+  maxDuration = 60000; //max duration of the buff
+  maxStacks = 12; //maximum stacks
```

Comment:
Since these are constants you should move them to a global variable instead of an instance variable